From 7d6b5293934cd1564ab4633eca09d0f03fe1cc4c Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Fri, 25 Oct 2024 13:53:47 +0800
Subject: [PATCH 238/277] fixup! mtk: mt76: mt7996: add Adv-TTLM support for
 STA

Send peer_mld mcu command only if it'a TTLM related event.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 mt7996/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mt7996/main.c b/mt7996/main.c
index b1e0288a..a716e4d8 100644
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -907,7 +907,8 @@ static void mt7996_vif_cfg_changed(struct ieee80211_hw *hw,
 		}
 	}
 
-	if (changed & BSS_CHANGED_MLD_VALID_LINKS)
+	if ((changed & BSS_CHANGED_MLD_VALID_LINKS) &&
+	    (changed & (BSS_CHANGED_MLD_ADV_TTLM | BSS_CHANGED_MLD_NEG_TTLM)))
 		mt7996_mcu_peer_mld_ttlm_req(dev, vif, changed);
 
 	mutex_unlock(&dev->mt76.mutex);
-- 
2.45.2

